From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Prathamesh Shete <pshete@nvidia.com>
Date: Fri, 30 Aug 2024 09:47:38 +0000
Subject: [PATCH 245/254] NVIDIA: SAUCE: gpio: tegra186: fix pinctrl for
 multi-socket

SFIO bit not getting cleared during PCIE dynamic switch
Current implementation picks pinctrl device node based on compatible
But as compatible is same for both socket0 and 1 nodes
Change this implementation and use phandle instead

Based upon the following commit and rebased on Linux v6.18:
https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3319388

Bug 4838834

Signed-off-by: Prathamesh Shete <pshete@nvidia.com>
[ Rebased by Jon Hunter on Linux v6.18 ]
Signed-off-by: Jon Hunter <jonathanh@nvidia.com>
Change-Id: I9ccb8766073fd7274262e5c8e0af734d2f6b60a3

diff --git a/drivers/gpio/gpio-tegra186.c b/drivers/gpio/gpio-tegra186.c
index e667e9f80bf6e..da07424bb6dce 100644
--- a/drivers/gpio/gpio-tegra186.c
+++ b/drivers/gpio/gpio-tegra186.c
@@ -478,7 +478,7 @@ static int tegra186_gpio_add_pin_ranges(struct gpio_chip *chip)
 	if (!gpio->soc->pinmux || gpio->soc->num_pin_ranges == 0)
 		return 0;
 
-	np = of_find_compatible_node(NULL, NULL, gpio->soc->pinmux);
+	np = of_parse_phandle(chip->parent->of_node, "pinctrl-node", 0);
 	if (!np)
 		return -ENODEV;
 
-- 
2.43.0

